UIPS/UIP-0131.md

```diff
@@ -0,0 +1,121 @@
+---
+uip: "0131"
+title: "Ministry"
+description: An Azimuth Multibridge
+author: ~master-malwyl, ~rolryx, ~sarlev-sarsen
+status: Draft
+type: Standards Track
+category: PKI
+created: ~2024.11.20
+---
+
+Abstract
+--------
+
+Ministry enables Urbit's Azimuth PKI to operate across multiple blockchain platforms while maintaining Ethereum as its root of trust. The system introduces a secure bridging protocol that allows points to move between supported chains through galaxy-operated oracles. This proposal specifies both the general bridge architecture and a concrete first implementation supporting Coinbase's Base layer-2 rollup.
+
+Motivation
+----------
+
+The Urbit namespace currently exists solely on Ethereum. While Ethereum provides strong property rights guarantees, limiting Azimuth to a single chain restricts user optionality and increases operational costs. As the blockchain ecosystem evolves with new Layer 2s and alternative Layer 1s, there is value in making Urbit IDs portable while preserving both the security guarantees and property rights of the existing system.
+
+Moving points to more cost-effective chains reduces operational overhead for users. Integration with multiple blockchain ecosystems increases accessibility and enables future compatibility with specialized Urbit-specific chains or features available on select global consensus ledgers other than mainnet Ethereum. The proposal maintains security through a galaxy-based oracle system with robust economic incentives.
+
+Base provides an ideal first target for cross-chain support due to its Ethereum compatibility, established security model, and lower transaction costs. This implementation will validate the Ministry protocol architecture while delivering immediate practical benefits to the network.
```

Comment:
> Base provides an ideal first target (...)
> This implementation will validate the Ministry protocol architecture

If the long-term intent is integration with a diverse range of targets, you probably want to start with integrating at least two or three very technically dissimilar targets, to prove out that your architecture can support all those different cases. Yes this increases scope, but reduces risk of unforeseen integration problems down the line.

Reply:
Most of your other objections seem to express concerns about scope; curious how you think about the tension between long term intent (including if you agree with it or not), and starting with reasonable size chunks and proving out the use cases (i.e. onboading users via Base and passkey wallets).